scipy/stats/_multivariate.py

```diff
@@ -1296,16 +1296,43 @@ def _lnB(alpha):
 
 _dirichlet_depr_message = (
 """  # noqa
-`dirichlet` is deprecated due to an interface inconsistency: compared to
-other distributions, methods `pdf` and `logpdf` expect the transpose of the
-input `x`. Please use `multivariate_beta`, which corrects this inconsistency.
+Methods `pdf` and `logpdf` of `dirichlet` are deprecated due to an interface
+inconsistency: compared to other distributions, methods `pdf` and `logpdf`
+expect the transpose of the input `x`. Please use `multivariate_beta`,
+which corrects this inconsistency.
 In SciPy 1.11.0, this deprecation warning will be removed and `dirichlet` will
 become an alias for `multivariate_beta`.
 """)
 
 
 class dirichlet_gen(multi_rv_generic):
-    r"""A Dirichlet random variable (deprecated, use `multivariate_beta` instead).
+    r"""A Dirichlet random variable.
+
+    .. deprecated:: 1.9.0
+        Methods `pdf` and `logpdf` of `dirichlet` are deprecated due to an
+        interface inconsistency: compared to other distributions, these
+        methods expect the transpose of the input `x`. Please use
+        `multivariate_beta`, which corrects this inconsistency.
+
+        `multivariate_beta` is a drop-in replacement for `dirichlet`
+        except that methods `pdf` and `logpdf` use a different convention.
+        For instance, code like this::
+
+            from scipy.stats import dirichlet
+            p = dirichlet.pdf(x)
+            logp = dirichlet.logpdf(x)
+
+        can be changed to::
+
+            from scipy.stats import multivariate_beta
+            p = multivariate_beta.pdf(np.transpose(x))
+            logp = multivariate_beta.logpdf(np.transpose(x))
+
+        All other methods of `multivariate_beta` and `dirichlet` are identical;
+        use of other methods  does not need to be changed.
+
+        In SciPy 1.11.0, this deprecation warning will be removed and
+        `dirichlet` will become an alias for `multivariate_beta`.
```

Comment:
I don't understand, are we deprecating the whole function or just two methods. The first line of the warning is misleading to me.

Reply:
This is not my idea. I'm trying to do something like what was suggested in https://github.com/scipy/scipy/pull/16042#issuecomment-1137227051.

It makes some sense to say that we are deprecating the _behavior_ of the `pdf` and `logpdf` methods. In 1.11, the behavior of those methods will change (to be the same as `multivariate_beta`). So at that point, the two names will be aliases of one another.

Reply:
Ok got it. But I am not sure I agree with doing something special here. I am never a fan of aliases like these. Users can have it on their end if they want to. IMO it's just confusing to have these in our codebase as it just increases the public API surface.